### PR TITLE
[Refactor][seatunnel-core-flink] refactor flink-sql module

### PR DESCRIPTION
**Pull request by legendtkl**
## Purpose of this pull request
refactor flink sql module, ref https://github.com/apache/incubator-seatunnel/issues/1753 and dev mail for some context

## Check list
* [yes] Code changed are covered with tests, or it does not need tests for reason:
* [yes] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [yes] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by legendtkl** [comment]
Hi, @ruanwenjun , would you help review this?

Just as we discussed in the dev mailing list, I send the PR for step 1, namely "refactor start-seatunnel-sql.sh. Now start-seatunnel-flink.sh and start-seatunnel-spark.sh have been refactored, and the main logic has
been rewritten by java code. I think we can first keep them consistent."

Thanks,
Kelu

File: seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        Common.setDeployMode(flinkCommandArgs.getDeployMode().getName());
+        this.appJar = Common.appLibDir().resolve(APP_JAR_NAME).toString();
+    }
+
+    @Override
+    public List<String> buildCommands() {
+        return CommandLineUtils.buildFlinkCommand(flinkCommandArgs, CLASS_NAME, appJar);
+    }
+
+    @SuppressWarnings("checkstyle:RegexpSingleline")
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java:51]:
```suggestion
```
Maybe this suppress is not needed here?

File: seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private static final String CLASS_NAME = SeatunnelSql.class.getName();
+
+
+    private final FlinkCommandArgs flinkCommandArgs;
+    /**
+     * SeaTunnel flink sql job jar.
+     */
+    private final String appJar;
+
+    public FlinkSqlStarter(String[] args) {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java]: Public is not needed here? If you want to use this class in UT, default is ok enough.
```suggestion
    FlinkSqlStarter(String[] args) {
```

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/config/FlinkJobStarter.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.core.flink.config;
+
+public enum FlinkJobStarter {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/config/FlinkJobStarter.java]: How about rename to FlinkJobType.

File: seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java [hunk 4]
```diff
@@ -35,4 +42,44 @@ public static FlinkCommandArgs parseFlinkArgs(String[] args) {
             .parse(args);
         return flinkCommandArgs;
     }
+
+    public static FlinkCommandArgs parseCommandArgs(String[] args, FlinkJobStarter starter) {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-flink/src/main/java/org/apache/seatunnel/core/flink/utils/CommandLineUtils.java]: You should directly enhance the method in line 37, now you can remove the method in line 37.

File: seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.seatunnel.core.flink.utils.CommandLineUtils;
+
+import java.util.List;
+
+public class FlinkSqlStarter implements Starter {
+
+    private static final String APP_JAR_NAME = "seatunnel-core-flink-sql.jar";
+    private static final String CLASS_NAME = SeatunnelSql.class.getName();
+
+
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java]: Remove the extra empty line.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]
Overall is ok to me, add some comments.

File: seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java (hunk 1 above)
Comment (legendtkl) [reply seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java:51]: Without this, we will get the style check fail "Prohibit invoking System.out.println in source code ! [RegexpSingleline]"

Comment (ruanwenjun) [reply seatunnel-core/seatunnel-core-flink-sql/src/main/java/org/apache/seatunnel/core/sql/FlinkSqlStarter.java:51]: OK, I forget this.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**